### PR TITLE
OLS-490: Refactor ModelConfig

### PR DESCRIPTION
**Pull request by onmete**
## Description

Refactor ModelConfig.
Next step is to add `model_parameters`, eg.:
```yaml
    models:
      - name: ibm/granite-13b-chat-v2
        context_window_size: 1000
        model_parameters:
           max_tokens_for_response: 20
```

## Type of change

- [x] Refactor

### CODE REVIEW

File: ols/app/models/config.py [hunk 1]
```diff
@@ -96,104 +97,61 @@ class InvalidConfigurationError(Exception):
...
-    name: Optional[str] = None
+    name: str
 
     # TODO: OLS-656 Switch OLS operator to use provider-specific configuration parameters
     url: Optional[AnyHttpUrl] = None
     credentials: Optional[str] = None
 
-    context_window_size: int = -1  # need to be set later, based on model
-    max_tokens_for_response: int = -1  # need to be set later, based on model
+    context_window_size: Optional[PositiveInt] = None
```
Comment (tisnik) [inline ols/app/models/config.py:106]: it is not optional, it is `constants.DEFAULT_CONTEXT_WINDOW_SIZE` by default

Comment (onmete) [reply ols/app/models/config.py:106]: Right.

**Comment by onmete** [comment]
/override "Red Hat Konflux / lightspeed-service-on-pull-request"

**Review by tisnik — COMMENTED** [review]
it looks good

**Comment by tisnik** [comment]
/override "Red Hat Konflux / lightspeed-service-on-pull-request"

**Pull request merged**